src/Symfony/Component/Validator/Validation.php

```diff
@@ -49,6 +49,26 @@ public static function createCallable($constraintOrValidator = null, Constraint
         };
     }
 
+    /**
+     * Creates a callable that returns true/false instead of throwing validation exceptions.
+     *
+     * @param Constraint|ValidatorInterface|null $constraintOrValidator
+     */
+    public static function createValidCallable($constraintOrValidator = null, Constraint ...$constraints): callable
+    {
+        $validator = self::createCallable(...func_get_args());
+
+        return static function ($value) use ($validator) {
```

Comment:
shouldn't we accept `&$e = null` as extra argument, to allow getting the error?

Reply:
i think this allows reverse logic as well,

`createValidatorCallable => bool`
`createCallable => throw`

the latter can obtain the exception then

Reply:
Maybe we could even add `&$violations` instead, and then we wouldn't have to rely on signaling by exception here: `createCallable` would call `createValidCallable`. Makes sense?

Reply:
to me, if you need the details, you should not be using the one returning a boolean. this callable is meant to provide an easy integrations with places like the OptionsResolver expecting a boolean anyway, which won't care about the exception returned by reference in an argument.
Btw, reference-based returned values in an argument are harder to discover.

Reply:
I'd still prefer removing signaling by exception...

Reply:
well, then maybe we should not build one of the callables on top of the other one, but make both callables (the one triggering an exception and the one returning a boolean) call the validator API for their need.

Reply:
the reference isn't bad per se :)

Reply:
well, in internal code, we can do what we want. But the returned callables are a public API

Reply:
or we just dup the bit of logic :)

Reply:
what's important to me; is a validator vs. the validation. A "valid callable" isnt a thing IMHO,

Reply:
I mean: having the reference as part if the public API would be useful to me (and also for internal needs if course, that's the proof of usefulness :))

Reply:
To me, if you need access to the violations, you should be using `$validator->validate()` directly instead of wrapping it in such callable.

Reply:
i tend to agree with @stof

```
try { createCallablle()('x') } 
catch (ValidationFailedException $e) { $e->getViolations() }
```

already works

Reply:
I understand that you both have a preference, but I don't see why we shouldn't add the reference.
It makes the API more useful, with no downside.

Reply:
Added `$violations`